ote_sdk/ote_sdk/tests/entities/test_model.py

```diff
@@ -0,0 +1,406 @@
+# Copyright (C) 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+from datetime import datetime
+import pytest
+from pathlib import Path
+import tempfile
+import os
+
+from ote_sdk.entities.model import (
+    ModelPrecision,
+    ModelStatus,
+    ModelFormat,
+    ModelOptimizationType,
+    OptimizationMethod,
+    ModelConfiguration,
+    ModelEntity
+)
+from ote_sdk.configuration import ConfigurableParameters
+from ote_sdk.entities.label_schema import LabelSchemaEntity
+from ote_sdk.entities.id import ID
+from ote_sdk.entities.model_template import TargetDevice
+from ote_sdk.tests.test_helpers import generate_random_single_image
+from ote_sdk.entities.image import Image
+from ote_sdk.entities.annotation import NullAnnotationSceneEntity
+from ote_sdk.entities.dataset_item import DatasetItemEntity
+from ote_sdk.entities.datasets import DatasetEntity
+from ote_sdk.utils.time_utils import now
+from ote_sdk.entities.model_template import parse_model_template
+from ote_sdk.configuration import ote_config_helper
+from ote_sdk.entities.task_environment import TaskEnvironment
+from ote_sdk.entities.label import LabelEntity, Domain
+from ote_sdk.entities.metrics import NullPerformance, Performance, ScoreMetric
+from ote_sdk.usecases.adapters.model_adapter import ModelAdapter
+
+from ote_sdk.tests.constants.ote_sdk_components import OteSdkComponent
+from ote_sdk.tests.constants.requirements import Requirements
+
+
+@pytest.mark.components(OteSdkComponent.OTE_SDK)
+class TestModelPrecision:
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_model_precision(self):
+        """
+        <b>Description:</b>
+        Check that ModelPrecision correctly return the precision name
+
+        <b>Expected results:</b>
+        Test passes if ModelPrecision correctly return the precision name
+
+        <b>Steps</b>
+        1. Check precisions in the ModelPrecision
+        """
+
+        model_precision = ModelPrecision
+        assert len(model_precision) == 4
+
+@pytest.mark.components(OteSdkComponent.OTE_SDK)
+class TestModelStatus:
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_model_status(self):
+        """
+        <b>Description:</b>
+        Check that ModelStatus correctly return the status name
```

Comment:
ditto

Reply:
Fixed here and in other places if it occurs.